### PR TITLE
Adjust logs for containerized and non-containerized services

### LINKED ISSUE #3835: fixpermissions: change rights on logs dir
On CentOS 7, initial permissions on logs files:
```console
# ls -rtl ../../logs/*log
-rw-r--r--  1 root pf         0 Nov 30 03:11 ../../logs/innobackup.log
-rw-r-----. 1 root pf         0 Nov 30 03:11 ../../logs/mariadb_error.log
-rw-rw-r--. 1 pf   pf         0 Nov 30 03:11 ../../logs/violation.log
-rw-r-----  1 root pf       102 Nov 30 09:16 ../../logs/pffilter.log
-rw-r-----. 1 root pf     34662 Nov 30 09:16 ../../logs/pfconfig.log
-rw-r-----  1 root pf       251 Nov 30 09:16 ../../logs/haproxy_db.log
-rw-r-----  1 root pf       767 Nov 30 09:16 ../../logs/pfsso.log
-rw-r-----  1 root pf       569 Nov 30 09:16 ../../logs/pfdns.log
-rw-r-----  1 root pf       673 Nov 30 09:16 ../../logs/haproxy_portal.log
-rw-r-----  1 root pf      2592 Nov 30 09:16 ../../logs/api-frontend.log
-rw-r-----  1 root pf      4247 Nov 30 09:17 ../../logs/radius.log
-rw-r-----  1 root pf      4247 Nov 30 09:17 ../../logs/radius-acct.log
-rw-r-----  1 root pf      4505 Nov 30 09:17 ../../logs/radius-load_balancer.log
-rw-rw-r--. 1 pf   pf       297 Nov 30 09:45 ../../logs/httpd.admin.audit.log
-rw-r-----  1 root pf      2543 Nov 30 09:46 ../../logs/httpd.admin.log
-rw-r-----  1 root pf     35157 Nov 30 09:56 ../../logs/redis_queue.log
-rw-r-----  1 root pf    154380 Nov 30 09:57 ../../logs/fingerbank.log
-rw-rw-r--. 1 pf   pf   1030843 Nov 30 09:57 ../../logs/packetfence.log
-rw-r-----  1 root pf    463954 Nov 30 09:58 ../../logs/pfstats.log
-rw-r-----. 1 root pf     88133 Nov 30 09:58 ../../logs/redis_server.log
-rw-rw-r--. 1 pf   pf     32804 Nov 30 09:58 ../../logs/redis_cache.log
-rw-r-----  1 root pf   1100387 Nov 30 09:58 ../../logs/pfmon.log
-rw-r-----  1 root pf 136398607 Nov 30 09:58 ../../logs/pfdhcplistener.log
```

After running `/usr/local/pf/bin/pfcmd fixpermissions`:
```console
# ls -rtl ../../logs/*log
-rw-rw-r--  1 root pf         0 Nov 30 03:11 ../../logs/innobackup.log
-rw-rw-r--. 1 root pf         0 Nov 30 03:11 ../../logs/mariadb_error.log
-rw-rw-r--. 1 pf   pf         0 Nov 30 03:11 ../../logs/violation.log
-rw-rw-r--  1 root pf       102 Nov 30 09:16 ../../logs/pffilter.log
-rw-rw-r--. 1 pf   pf     34662 Nov 30 09:16 ../../logs/pfconfig.log
-rw-rw-r--  1 root pf       251 Nov 30 09:16 ../../logs/haproxy_db.log
-rw-rw-r--  1 root pf       767 Nov 30 09:16 ../../logs/pfsso.log
-rw-rw-r--  1 pf   pf       569 Nov 30 09:16 ../../logs/pfdns.log
-rw-rw-r--  1 root pf       673 Nov 30 09:16 ../../logs/haproxy_portal.log
-rw-rw-r--  1 root pf      2592 Nov 30 09:16 ../../logs/api-frontend.log
-rw-rw-r--  1 root pf      4247 Nov 30 09:17 ../../logs/radius.log
-rw-rw-r--  1 root pf      4247 Nov 30 09:17 ../../logs/radius-acct.log
-rw-rw-r--  1 root pf      4505 Nov 30 09:17 ../../logs/radius-load_balancer.log
-rw-rw-r--. 1 pf   pf       297 Nov 30 09:45 ../../logs/httpd.admin.audit.log
-rw-rw-r--  1 pf   pf      2543 Nov 30 09:46 ../../logs/httpd.admin.log
-rw-rw-r--  1 root pf     35157 Nov 30 09:56 ../../logs/redis_queue.log
-rw-rw-r--  1 root pf    154380 Nov 30 09:57 ../../logs/fingerbank.log
-rw-rw-r--. 1 pf   pf   1030843 Nov 30 09:57 ../../logs/packetfence.log
-rw-rw-r--. 1 root pf     88133 Nov 30 09:58 ../../logs/redis_server.log
-rw-rw-r--. 1 pf   pf     32804 Nov 30 09:58 ../../logs/redis_cache.log
-rw-rw-r--  1 root pf    464117 Nov 30 09:58 ../../logs/pfstats.log
-rw-rw-r--  1 pf   pf   1102208 Nov 30 09:58 ../../logs/pfmon.log
-rw-rw-r--  1 pf   pf 136502707 Nov 30 09:58 ../../logs/pfdhcplistener.log
```

Due to https://github.com/inverse-inc/packetfence/commit/28f8f621bb84059f1b05d3e11c77f4e295e1249e, do you think we should keep `fixpermissions` on log dir ?

### PR DESCRIPTION
**Pull request by nqb**
# Description
- rsyslog will look into systemd-journal to get logs for containerized and non-containerized services
- disable `docker logs` for all containerized-services: everything is in journald
- adjust rsyslog rules for services (old, new, containerized)
- remove management of permissions on log directory and files using `fixpermissions`: 
  * log directory is installed by packaging
  * rsyslog will create/update files inside log directory (owned by `root` and group `pf`)
  * logrotate will handle log rotation
  * `pfcmd checkup` ensure rights are correct on log files
* `innobackup.log` is handle separately

# Issues
fixes #3835

# Impacts
- rsyslog
- logrotate
- permissions on logs/ and log files

# Delete branch after merge
YES

# NEWS file entries
## Enhancements
* Adjus logs for containerized and non-containerized services (#FIXME)

# UPGRADE file entries

```asciidoc
=== Log files names updated

The name of some log files have changed. You can find a list below:

.Mapping between old and new log files
|===
|Service |Old log file(s) |New log file(s)

|MariaDB
|mariadb_error.log
|mariadb.log

|httpd.aaa (Apache requests)
|httpd.aaa.access and httpd.aaa.error
|httpd.apache

|httpd.collector (Apache requests)
|httpd.collector.log and httpd.collector.error
|httpd.apache

|httpd.portal (Apache requests)
|httpd.portal.access, httpd.portal.error, httpd.portal.catalyst
|httpd.apache

|httpd.proxy (Apache requests)
|httpd.proxy.error and httpd.proxy.access
|httpd.apache

|httpd.webservices (Apache requests)
|httpd.webservices.error and httpd.webservices.access
|httpd.apache

|api-frontend (Apache requests)
|httpd.api-frontend.access 
|httpd.apache

|HAProxy (all services)
|/var/log/syslog or /var/log/messages
|haproxy.log


If you use a Syslog Forwarder to send log files to a remote Syslog server, you should run following script:

[code,bash]
----
/usr/local/pf/addons/upgrade/to-12.0-rename-log-files.pl
----
```

### CODE REVIEW

**Comment by nqb** [comment]
I ran following pipeline to start testing with new packages: https://gitlab.com/inverse-inc/packetfence/-/pipelines/575065134

File: lib/pf/constants/syslog.pm [hunk 1]
```diff
@@ -103,11 +57,16 @@ our @SyslogInfo = (
...
         'name'       => 'packetfence.log',
         'conditions' => [
-            '$syslogtag contains "packetfence"',
-            '$programname == "pfqueue"',
-            '($programname == "pfhttpd" and not $msg contains "GET /api/v1/logs/tail/")',
+            '$programname contains "packetfence"',
+            '$programname == "pfqueue-docker-wrapper"',
+            '($syslogtag == "pfhttpd" and not $msg contains "GET /api/v1/logs/tail/")',
             '$programname == "pfipset"',
-            '$programname == "pfpki"',
```
Comment (JeGoi) [inline lib/pf/constants/syslog.pm:110]: Why this one is removed? and not change to the wrapper too?

Comment (nqb) [reply lib/pf/constants/syslog.pm:110]: Replaced by `pfpki-docker-wrapper`

Comment (JeGoi) [reply lib/pf/constants/syslog.pm:110]: Yes I just saw it bellow. Cool

**Review by JeGoi — APPROVED** [review]
It looks fine for me.
Questions answered.

**Comment by nqb** [comment]
I finished to fix some issues. Running a new pipeline (https://gitlab.com/inverse-inc/packetfence/-/pipelines/576942480) to test new packages.

**Comment by nqb** [comment]
Previously, it was possible to forward or live view `/var/log/syslog` or `/var/log/messages` (depending on OS). Currently, it doesn't work due to https://github.com/inverse-inc/packetfence/issues/7049#issuecomment-1174726034.

Also, it would mean we will need to expose `/var/log` inside `api-frontend` container. Not sure this is what we want.

**Comment by julsemaan** [comment]
> Previously, it was possible to forward or live view `/var/log/syslog` or `/var/log/messages` (depending on OS). Currently, it doesn't work due to [#7049 (comment)](https://github.com/inverse-inc/packetfence/issues/7049#issuecomment-1174726034).
> 
> Also, it would mean we will need to expose `/var/log` inside `api-frontend` container. Not sure this is what we want.

I don't think we need to live view the syslog file anymore as long as we make sure all the PF services log in files in /usr/local/pf/logs. Historically, some services like keepalived and haproxy didn't log there and that's why that file can be viewed in the log viewer. 

We could also mount this directory in read-only on the api-frontend container

**Comment by nqb** [comment]
I added a new rule to redirect `[GIN]` messages (related to `fingerbank-collector`) into `fingerbank.log`. They previously go into `/var/log/{syslog,messages}`. However, there is still some messages related to `fingerbank-collector` which go into `/var/log/{syslog,messages}` but they are not easy to distinguish from others.

Example:

```
Jul 05 10:53:34 pfdeb11dev bash[17072]: export PORT=4723
Jul 05 10:53:34 pfdeb11dev bash[17072]: export COLLECTOR_ENDPOINTS_CACHE_PATH=/usr/local/fingerbank/db/collector_endpoints_cache.db
Jul 05 10:53:34 pfdeb11dev bash[17072]: export COLLECTOR_DB_PERSISTENCE_INTERVAL=60s
Jul 05 10:53:34 pfdeb11dev bash[17072]: export COLLECTOR_IP_MAPS_DB_PATH=/usr/local/fingerbank/db/collector_ip_maps.db
Jul 05 10:53:34 pfdeb11dev bash[17072]: export COLLECTOR_CLUSTERED=true
Jul 05 10:53:34 pfdeb11dev bash[17072]: export COLLECTOR_CLUSTER_RESYNC_INTERVAL=120s
Jul 05 10:53:34 pfdeb11dev bash[17072]: export COLLECTOR_CONTACT_PEERS=
Jul 05 10:53:34 pfdeb11dev bash[17072]: export GOGC=100
Jul 05 10:53:34 pfdeb11dev bash[17072]: export COLLECTOR_DELETE_INACTIVE_ENDPOINTS=168h
```
I don't think it's critical to have this ones in `fingerbank.log` but I remembered to see some "monitoring" messages related to bash scripts coming from `fingerbank-collector`  in `/var/log/{syslog,messages}`.

**Comment by nqb** [comment]
@julsemaan, I'm talking about:
```
Jul  5 11:55:35 pfdeb11dev bash[1315]: Failed check cb8f0450-e1b7-11e7-80c1-9a214cf093ae:w
Jul  5 11:55:35 pfdeb11dev bash[1315]: The collector doesn't seem to be running alongside PacketFence.
```

**Comment by nqb** [comment]
Unless the `/var/log/{syslog,messages]` decision, nothing more to add on my side. Tested on Debian and EL systems

**Comment by nqb** [comment]
As discussed internally, I exposed `/var/log` directory inside `api-frontend` Docker container in read-only. It's hard to expose only `/var/log/messages` or `/var/log/syslog` at the moment before we handle properly `os_detection` inside containers.

Please review and merge if everything is good for you.

**Comment by julsemaan** [comment]
@nqb, do you think we could have an upgrade note we can add that would state what changed to where. Perhaps not for all the smaller services but like saying mariadb_error.log is in mariadb.log looks important enough to be included

File: docs/installation/best_practices.asciidoc [hunk 2]
```diff
@@ -24,7 +24,7 @@ IPTables is now entirely managed by PacketFence. However, if you need to perform
 
 === Log Rotations
 
-PacketFence can generate a lot of log entries in huge production environments. This is why we recommend to use `logrotate` to periodically rotate your logs. A working logrotate script is provided with the PacketFence package. This script is located under the `/usr/local/pf/packetfence.logrotate` file, and it's configured to do a daily log rotation and keeping old logs with compression. It has been added during PacketFence initial installation.
+PacketFence can generate a lot of log entries in huge production environments. This is why we recommend to use `logrotate` to periodically rotate your logs. A working logrotate script is provided with the PacketFence package. This script is located inside logrotate directory, and it's configured to do a daily log rotation and keeping old logs with compression. It has been added during PacketFence initial installation.
```
Comment (julsemaan) [inline docs/installation/best_practices.asciidoc]:
```suggestion
PacketFence can generate a lot of log entries in huge production environments. This is why we recommend to use `logrotate` to periodically rotate your logs. A working logrotate script is provided with the PacketFence package. This script is located inside the logrotate directory (`/etc/logrotate.d/`), and it's configured to do a daily log rotation and keeping old logs with compression. It has been added during PacketFence initial installation.
```

**Comment by nqb** [comment]
> @nqb, do you think we could have an upgrade note we can add that would state what changed to where. Perhaps not for all the smaller services but like saying mariadb_error.log is in mariadb.log looks important enough to be included

100% agree, will do that tomorrow morning first thing.

**Comment by nqb** [comment]
In fact, I already add an upgrade note in PR description but it can be improved.

**Comment by nqb** [comment]
Upgrade note updated and I also push a small fix to avoid keeping old `httpd.*.access,error,catalyst` files after upgrade.

**Comment by nqb** [comment]
@JeGoi and @julsemaan, could you merge if all is fine ?
Thanks!

**Review by JeGoi — APPROVED** [review]
Wow looks great!

**Pull request merged**